src/shape.cpp

```diff
@@ -104,10 +125,13 @@ Dimension Dimension::any() {
 Shape::Shape() {
 }
 
-Shape::Shape(std::initializer_list<Dimension> list) {
-    this->dimensions.reserve(list.size());
-    for (const Dimension& dim : list) {
-        this->dimensions.emplace_back(dim);
+Shape::Shape(std::initializer_list<Dimension> list) :
+    std::vector<Dimension>(list) {}
+
+Shape::Shape(const shape_t& s) {
+    auto status = fromFlatShape(s, *this);
+    if (!status.ok()) {
+        throw 42;  // TODO this constructor is to be removed during 1.0 API cleanup
```

Comment:
Why during cleanup? Where is it needed now?

Reply:
Most constructors of TensorInfo convert shape_t to Shape. We need to remove those first.